pwiz_tools/Skyline/FileUI/BaseFileDialogNE.cs

```diff
@@ -0,0 +1,1194 @@
+﻿/*
+ * Original author: Matt Chambers <matt.chambers .@. vanderbilt.edu>
+ *
+ * Copyright 2009 Vanderbilt University - Nashville, TN 37232
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Windows.Forms;
+using System.IO;
+using pwiz.Common.Collections;
+using pwiz.Common.SystemUtil;
+using pwiz.Skyline.Alerts;
+using pwiz.Skyline.Model.Results;
+using pwiz.Skyline.Model.Results.RemoteApi;
+using pwiz.Skyline.Properties;
+using pwiz.Skyline.Util;
+using pwiz.Skyline.Util.Extensions;
+
+
+namespace pwiz.Skyline.FileUI
+{
+    public partial class BaseFileDialogNE : FormEx
+    {
+        private readonly ListViewColumnSorter _listViewColumnSorter = new ListViewColumnSorter();
+        private readonly Stack<MsDataFileUri> _previousDirectories = new Stack<MsDataFileUri>();
+        private readonly int _specialFolderCount;
+        private readonly int _myComputerIndex;
+        private readonly int _remoteIndex;
+        private RemoteSession _remoteSession;
+        private readonly IList<RemoteAccount> _remoteAccounts;
+        private bool _waitingForData;
+        private readonly IList<string> _specificDataSourceFilter; // Specific data sources to look for
+        
+        /// <summary>
+        /// File picker which is aware of mass spec "files" that are really directories
+        /// </summary>
+        /// <param name="remoteAccounts">For UNIFI</param>
+        /// <param name="specificDataSourceFilter">Optional list of specific files the user needs to located, ignoring the rest</param>
+        public BaseFileDialogNE(IList<RemoteAccount> remoteAccounts, IList<string> specificDataSourceFilter = null)
+        {
+            InitializeComponent();
+            _remoteAccounts = remoteAccounts;
+
+            listView.ListViewItemSorter = _listViewColumnSorter;
+
+            DialogResult = DialogResult.Cancel;
+
+            string[] sourceTypes =

```

Comment:
`sourceTypes` should become a ctor parameter, and this default list should move to OpenDataSourceDialog. I guess to pass it in to the base() keyword it may need to be a static member in OpenDataSourceDialog rather than a temporary array in the ctor body. So OpenFileDialogNE.ctor will take a `sourceTypes` parameter, OpenDataSourceDialog.ctor will not (it will pass its static member as the sourceTypes).